application/ui/components/input/array.jsx

```diff
@@ -0,0 +1,147 @@
+/** @jsx React.DOM */
+'use strict';
+
+var _             = require('underscore');
+var React         = require('react');
+
+module.exports = React.createClass({
+
+    displayName : 'ArrayParameter',
+
+    getInitialState : function()
+    {
+        return {
+            values   : [],
+            numItems : 0
```

Comment:
Actually that's just the key in the array that's being passed to `getInput`. It's not `this.state.numItems`.